client/state/jetpack-connect/test/selectors.js

```diff
@@ -6,49 +6,257 @@ import { expect } from 'chai';
 /**
  * Internal dependencies
  */
-import { isCalypsoStartedConnection, getFlowType, getJetpackSiteByUrl, hasXmlrpcError } from '../selectors';
-
-const stateHasXmlrpcError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'transport error - HTTP status code was not 200 (502)'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
-
-const stateHasNoError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: false
-		}
-	}
-};
-
-const stateHasNoAuthorizationCode = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Could not verify your request.'
-			}
-		}
-	}
-};
-
-const stateHasOtherError = {
-	jetpackConnect: {
-		jetpackConnectAuthorize: {
-			authorizeError: {
-				message: 'Jetpack: [already_connected] User already connected.'
-			},
-			authorizationCode: 'xxxx'
-		}
-	}
-};
+import {
+	getConnectingSite,
+	getAuthorize,
+	getAuthorizeQuery,
+	getAuthorizeSite,
+	getSessions,
+	getSSOSessions,
+	getSSO,
+	isCalypsoStartedConnection,
+	getFlowType,
+	getJetpackSiteByUrl,
+	hasXmlrpcError
+} from '../selectors';
 
 describe( 'selectors', () => {
+	describe( '#getConnectingSite()', () => {
+		it( 'should return undefined if user has not started connecting a site', () => {
+			const state = {
+				jetpackConnect: {}
+			};
+
+			expect( getConnectingSite( state ) ).to.be.undefined;
+		} );
+
+		it( 'should return the current connecting site if there is such', () => {
+			const jetpackConnectSite = {
+				url: '',
+				isFetching: true,
+				isFetched: false,
+				isRedirecting: false,
+				installConfirmedByUser: true,
+				isDismissed: false,
+				data: {
+					isDotCom: false,
+					notExists: false,
+					notWordPress: false,
+					notJetpack: false,
+					jetpackVersion: '4.3.1',
+					outdatedJetpack: false,
+					notActiveJetpack: false,
+					notConnectedJetpack: true,
+					alreadyOwned: true,
+					alreadyConnected: false
+				}
+			};
+			const state = {
+				jetpackConnect: {
+					jetpackConnectSite
+				}
+			};
+
+			expect( getConnectingSite( state ) ).to.eql( jetpackConnectSite );
```

Comment:
I get your point and you're perfectly right, but there must be a reason why we're not doing that anywhere in our selectors. @aduth @mtias do you guys have any idea why we return a reference to the state in our selectors, and not a copy?

Anyway I think this is outside of the scope of this PR, as it would affect a lot of other selectors throughout Calypso as well.